features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
Thanks @pan- for providing the answers.

>    Since these both solve the same problem of global-initialization, shouldn't we chose one solution and consolidate?

In this manner, yes they are the same, but from the perspective of singleton, they are not, thus depends on the usage. Therefore I proposed a name change for SingletonPtr, not to confuse that this class is not a singleton. As @pan- wrote:

> Wrapping the variable inside a LazyInstance or a function scope (as static) is the only option here (both have their advantages and inconvenient). 

As there's PR for providing locks for local static variables, any other concerns?